src/Raven.Studio/typescript/models/resources/info/resourceInfo.ts

```diff
@@ -74,6 +75,25 @@ abstract class resourceInfo {
 
     abstract asResource(): resource;
 
+    abstract update(resourceInfo: Raven.Client.Data.ResourceInfo): void;
+    
+    updateCurrentInstance(dto: Raven.Client.Data.ResourceInfo) {
```

Comment:
why we have both update and updatecurrent instance? Can't we have update in resource info, and call super.update in databaseinfo or filesysteminfo class?